### ISSUE TITLE
Generation of rustlexer.cpp silently fails

### ISSUE THREAD

**Issue by tomtab** [cgdb/cgdb#123]
make is failing for me at the moment (HEAD is 8debd2d) with a silent failure of the ylwrap script:
Here's the output:
make  all-recursive
make[1]: Entering directory `../cgdb'
Making all in lib
make[2]: Entering directory `../cgdb/lib'
Making all in util
make[3]: Entering directory `../cgdb/lib/util'
make[3]: Nothing to be done for `all'.
make[3]: Leaving directory `../cgdb/lib/util'
Making all in rline
make[3]: Entering directory `../cgdb/lib/rline'
make[3]: Nothing to be done for `all'.
make[3]: Leaving directory `../cgdb/lib/rline'
Making all in adt
make[3]: Entering directory `../cgdb/lib/adt'
make[3]: Nothing to be done for `all'.
make[3]: Leaving directory `../cgdb/lib/adt'
Making all in tgdb
make[3]: Entering directory `../cgdb/lib/tgdb'
make[3]: Nothing to be done for `all'.
make[3]: Leaving directory `../cgdb/lib/tgdb'
Making all in kui
make[3]: Entering directory `../cgdb/lib/kui'
make[3]: Nothing to be done for `all'.
make[3]: Leaving directory `../cgdb/lib/kui'
Making all in tokenizer
make[3]: Entering directory `../cgdb/lib/tokenizer'
/bin/sh ../../config/ylwrap rustlexer.lpp lex.yy.c rustlexer.cpp -- flex  
make[3]: *** [rustlexer.cpp] Error 1
make[3]: Leaving directory `../cgdb/lib/tokenizer'
make[2]: *** [all-recursive] Error 1
make[2]: Leaving directory `../cgdb/lib'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `../cgdb'
make: *** [all] Error 2

I ran the script manually and there weren't any errors, but it didn't generate the rustlexer.cpp file.
I am using flex 2.5.35 on CentOS 6.8.

Do you know how I can fix this ?
Is this because my flex is too old ?

It would be nice to be able to skip generating the rust lexer through configure or make,
but I am not enthusiastic about adding this myself (although I have a feeling I will have to, in the end).

Any help is appreciated.

**Comment by tomtab** [comment]
Just noticed that the FAQ file mentions flex 2.5.4, but it does not say that it is a requirement.
It's not a good sign, though.

**Comment by tomtab** [comment]
I installed flex 2.6.3 (latest release) and I'm getting the same silent failure.
So it's not flex's fault.

**Comment by brasko** [comment]
I can't reproduce this issue, sorry. What do you think the exact cause of the error is? The log you posted doesn't really show that. If it helps, I'm using flex 2.6.0.

**Comment by tomtab** [comment]
Victory! I got it to work.

It turns out that flex was not the problem.
Apparently, the problem was with the version of ylwrap from the default automake for CentOS 6.8 (1.11.1). I tried using the ylwrap from automake 1.15 and it worked (just replaced the symlink).

I know it wasn't flex's fault because I switched back to flex 2.5.35 to see if it would work with the newer ylwrap and it did.

However, I hit another error after this, but from g++ this time:
/usr/bin/g++ -DHAVE_CONFIG_H -DPKGDATADIR=\"/home/tmt/bin/share/cgdb\" -DTOPBUILDDIR=\"/home/tmt/cgdb-official\" -I. -I..    -I../lib/adt -I../lib/kui -I../lib/rline -I../lib/util -I../lib/tgdb -I../lib/tokenizer -g -O2 -MT cgdbrc.o -MD -MP -MF .deps/cgdbrc.Tpo -c -o cgdbrc.o cgdbrc.cpp
cgdbrc.cpp:69: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:70: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:71: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:72: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:73: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:74: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:76: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:77: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:78: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:80: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:81: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:82: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:83: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:84: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:85: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:86: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:87: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:88: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:89: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:90: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:91: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:92: error: expected primary-expression before ‘.’ token
cgdbrc.cpp:93: error: expected primary-expression before ‘.’ token

I thought I would try clang++ instead, since I was pretty sure it would have better C++ support than the default g++, and I was right!
So, the default g++ from CentOS 6.8 (4.4.7) can't build the latest cgdb, but the EPEL clang++ (3.4.2) can.

I can now successfully build the latest cgdb.
I hope all of this information will be helpful for someone.

Anyway, thanks for the quick reply.
Oh, and thanks for working on cgdb. It's a really helpful tool!
